pyiceberg/io/fsspec.py

```diff
@@ -121,9 +121,19 @@ def __call__(self, request: "AWSRequest", **_: Any) -> None:
         signer_url = self.properties.get(S3_SIGNER_URI, self.properties[URI]).rstrip("/")  # type: ignore
         signer_endpoint = self.properties.get(S3_SIGNER_ENDPOINT, S3_SIGNER_ENDPOINT_DEFAULT)
 
-        signer_headers = {}
+        signer_headers: dict[str, str] = {}
+
+        auth_header: str | None = None
         if token := self.properties.get(TOKEN):
-            signer_headers = {"Authorization": f"Bearer {token}"}
+            auth_header = f"Bearer {token}"
+        elif auth_manager := self.properties.get(AUTH_MANAGER):
+            header = getattr(auth_manager, "auth_header", None)
+            if callable(header):
+                auth_header = header()
```

Comment:
```suggestion
            auth_header = auth_manager.auth_header()
```
could we just call the function directly? this will fail if the `auth_header` function does not exist. 
i think the current solution will fail silently, i.e. not add any auth header if the `auth_header` function does not exist